Source/IGListAdapter.m

```diff
@@ -563,14 +564,13 @@ - (void)updateObjects:(NSArray *)objects dataSource:(id<IGListAdapterDataSource>
     IGParameterAssert(dataSource != nil);
 
 #if DEBUG
-    NSCountedSet *identifiersSet = [NSCountedSet new];
     for (id object in objects) {
-        [identifiersSet addObject:[object diffIdentifier]];
         IGAssert([object isEqualToDiffableObject:object], @"Object instance %@ not equal to itself. This will break infra map tables.", object);
-        IGAssert([identifiersSet countForObject:[object diffIdentifier]] <= 1, @"Diff identifier %@ for object %@ occurs more than once. Identifiers must be unique!", [object diffIdentifier], object);
     }
 #endif
 
+    NSArray *uniqueObjects = objectsWithDuplicateIdentifiersRemoved(objects);
```

Comment:
While not opposed to this change, this does make [deduping in the updater](https://github.com/Instagram/IGListKit/blob/master/Source/IGListAdapterUpdater.m#L109) redundant.

Maybe instead the deduping should be done here:

https://github.com/Instagram/IGListKit/blob/master/Source/IGListAdapter.m#L146

Reply:
@rnystrom If I dedupe at https://github.com/Instagram/IGListKit/blob/master/Source/IGListAdapter.m#L146 we still have the same issue reappear on reloadDataWithCompletion for adapter: https://github.com/Instagram/IGListKit/blob/b3ffafc018978c20845a3b5aa7b2edb9db8c7622/Source/IGListAdapter.m#L368

Reply:
Oooooooh, now I see it! Wo this is an awesome find @yemodin! How about we:

- Move the dedupe to [public settings change](https://github.com/Instagram/IGListKit/blob/master/Source/IGListAdapter.m#L146) **_and_** [reloadData](https://github.com/Instagram/IGListKit/blob/master/Source/IGListAdapter.m#L362)
- We make a followup issue to consolidate and dedupe in one place

I actually want to move object generation entirely so that "to" object arrays are only constructed once. Atm they are built w/ every call to `performUpdatesAnimated:completion:`, even if a bunch of calls are updated into a single batch.

Reply:
Done. Also made deduping before https://github.com/Instagram/IGListKit/blob/b3ffafc018978c20845a3b5aa7b2edb9db8c7622/Source/IGListAdapter.m#L322